### PR TITLE
Parameter Namespacing: Refactoring using the ParameterPath object.

### PR DESCRIPTION
**Pull request by AAlon**
Related change: https://github.com/aws-robotics/utils-ros1/pull/3

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws_common/include/aws_common/sdk_utils/parameter_reader.h [hunk 1]
```diff
@@ -20,6 +20,114 @@
...
+     *  2. Detailed lists of strings were specified for the parameter & node namespaces.
+     *    In this case, we construct the resolved path using the provided separators. Conversion would not apply.
+     * @param node_namespace_separator
+     * @param parameter_namespace_separator
+     * @param convert_node_ns_to_parameter_ns Only applies if the path was specified as a single string,
+     *   and the string contains node separators but does not contain any parameter separators.
+     * @return string representing the full, resolved path of the parameter.
+     * @note If node_namespaces and parameter_path_keys are empty, an empty string would be returned.
+     */
+    std::string get_resolved_path(char node_namespace_separator,
```
Comment (ross-desmond) [inline aws_common/include/aws_common/sdk_utils/parameter_reader.h]: Is this intended to be able to convert ROS2->ROS1 paths as well? If not we should specify that in the constructor  and deprecate it. This get_resolved_path almost is able to do both, and could be abused that way.
Also why don't we just put get_resolved_path within the rosparamreader itself, that way the param reader can choose how to format the parameter path?

File: aws_common/include/aws_common/sdk_utils/parameter_reader.h [hunk 2]
```diff
@@ -20,6 +20,114 @@
...
+          std::string resolved_path = get_node_path(node_namespace_separator);
+          if (!resolved_path.empty()) {
+              resolved_path += node_namespace_separator;
+          }
+          resolved_path += get_local_path(parameter_namespace_separator);
+          return resolved_path;
+      }
+    }
+private:
+    const std::vector<std::string> parameter_path_keys_;
```
Comment (ross-desmond) [inline aws_common/include/aws_common/sdk_utils/parameter_reader.h]: Comments here

File: aws_common/include/aws_common/sdk_utils/parameter_reader.h [hunk 3]
```diff
@@ -20,6 +20,110 @@
...
+private:
+    /**
+     * Member variables to store the parameter's path in the namespace hierarchy.
+     * parameter_path_keys_ Parameter namespaces list.
+     * node_namespaces_ Node namespaces list.
+     * name_ A single string. Used when simpler initialization is required, instead of the aforementioned namespace lists.
+     */
+    const std::vector<std::string> parameter_path_keys_;
+    const std::vector<std::string> node_namespaces_;
+    mutable std::string name_;
```
Comment (mm318) [inline aws_common/include/aws_common/sdk_utils/parameter_reader.h]: This doesn't need to be mutable anymore.

**Comment by AAlon** [comment]
Addressed feedback.

**Pull request merged**